### PR TITLE
Improve DeveloperGuide Formatting

### PR DESCRIPTION
**Pull request by haoyangw**
Improvements:
-Added complete `@@author` tags to attribute correct credit in RepoSense
-Fix section and subsection headers and update Table of Contents
-Fill in empty `Saving Data` section of DeveloperGuide
-Re-arrange sections to follow the layout of the [reference DeveloperGuide](https://se-education.org/addressbook-level3/DeveloperGuide.html)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -516,12 +523,15 @@ Then, it will call the `runCommands()` method to process the user input and exec
...
+<!-- @@author -->
 
-### Managers
+<!-- @@author haoyangw -->
+## Managers
 
 ![Manager](diagrams/Manager.png)<br/>
 The `Manager` abstract class is the superclass for classes responsible for handling user interaction with the app.
 
+### Implementation Rationale
```
Comment (wenxin-c) [inline docs/DeveloperGuide.md:534]: Should we standardise as "Design Considerations" like what other sections used?

**Review by wenxin-c — COMMENTED** [review]
Minor suggestions for your consideration!!

**Review by wenxin-c — APPROVED** [review]
Minor suggestions for your consideration!!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (haoyangw) [reply docs/DeveloperGuide.md]: Ok fixed! thanks

**Pull request merged**